### PR TITLE
Multiple Memory Server Implementation phase 3

### PR DESCRIPTION
**Pull request by gautham-bhat-k**
1] Added RPC interface for memory service
2] CIS main program

### CODE REVIEW

File: src/cis/fam_cis_async_handler.h [hunk 1]
```diff
@@ -83,24 +82,25 @@ typedef Fam_CIS_Rpc::WithAsyncMethod_copy<Fam_CIS_Server> sType;
...
-        service->cis_server_initialize(name, libfabricPort, provider, famCIS);
+        service->cis_server_initialize(famCIS);
     }
 
     ~Fam_CIS_Async_Handler() { delete service; }
 
     void dump_profile() {
         CIS_ASYNC_PROFILE_DUMP();
-        service->dump_profile();
+        // service->dump_profile();
```
Comment (mashoodk) [inline src/cis/fam_cis_async_handler.h:98]: commented code ?

File: src/common/fam_ops_libfabric.h [hunk 2]
```diff
@@ -78,15 +80,20 @@ class Fam_Ops_Libfabric : public Fam_Ops {
...
-    Fam_Ops_Libfabric(MemServerMap name, const char *service, bool is_source,
+    Fam_Ops_Libfabric(bool is_source, const char *provider,
+                      Fam_Thread_Model famTM, Fam_Allocator_Client *famAlloc,
+                      Fam_Context_Model famCM = FAM_CONTEXT_DEFAULT);
+    Fam_Ops_Libfabric(bool source, const char *libfabricProvider,
+                      Fam_Thread_Model famTM, Fam_Allocator_Client *famAlloc,
+                      Fam_Context_Model famCM, const char *memServerName,
+                      const char *libfabricPort);
+#if 0
+    Fam_Ops_Libfabric(Server_Map name, const char *service, bool is_source,
```
Comment (mashoodk) [inline src/common/fam_ops_libfabric.h:92]: commented code ?

File: test/unit-test/fam-api/fam_invalid_key_test.cpp [hunk 3]
```diff
@@ -59,35 +59,39 @@ int main() {
     // Initialize the libfabric ops
     char *message = strdup("This is the datapath test");
     char *name = strdup("127.0.0.1");
-    char *service = strdup(TEST_LIBFABRIC_PORT);
+    // char *service = strdup(TEST_LIBFABRIC_PORT);
```
Comment (mashoodk) [inline test/unit-test/fam-api/fam_invalid_key_test.cpp:62]: commented code.. (also in other places in this file)

File: test/common/fam_test_config.h.in [hunk 4]
```diff
@@ -63,12 +66,13 @@ using namespace openfam;
 void init_fam_options(Fam_Options *famOpts) {
     memset((void *)famOpts, 0, sizeof(Fam_Options));
 
-    famOpts->memoryServer = strdup(TEST_MEMORY_SERVER);
+	famOpts->openFamModel = strdup(TEST_OPENFAM_MODEL);
```
Comment (mashoodk) [inline test/common/fam_test_config.h.in:69]: spacing ?

File: src/common/fam_internal.h [hunk 5]
```diff
@@ -117,6 +117,8 @@ namespace openfam {
 #define DATAITEMID_MASK ((1UL << DATAITEMID_BITS) - 1)
 #define DATAITEMID_SHIFT 1
 
+#define ADDR_SIZE 20
```
[the remark this answers is not in the document]
Comment (gautham-bhat-k) [reply src/common/fam_internal.h:120]: This size is size of char array to hold ip address

**Review by mashoodk — APPROVED** [review]

**Review by VrashiPonnappa — APPROVED** [review]

**Pull request merged**